src/Blueprint/Blueprint/Cmdlets/BlueprintDefinition/BlueprintDefinitionCmdletBase.cs

```diff
@@ -149,8 +150,17 @@ protected string CreateFolderIfNotExist(string path, string folderName)
         {
             var folderPath = Path.Combine(path, folderName);
 
-            if (!AzureSession.Instance.DataStore.DirectoryExists(folderPath))
+            if (AzureSession.Instance.DataStore.DirectoryExists(folderPath))
             {
+                AzureSession.Instance.DataStore.EmptyDirectory(folderPath);
```

Comment:
@filizt is there any reason we want to delete all of the files in this user's folder?

Reply:
We wanted to make sure the the latest exported blueprint and its artifacts are the only blueprint and artifacts in the folder. So, if the user exported a blueprint and then made some changes(say deleted artifacts) through the portal and exported again, in the current implementation the export operation will keep the "deleted" artifact in the Artifacts folder since we are not doing any diffing between the local copy and the portal version of the blueprint. So, we thought we could delete the files in this case and re-create the blueprint and its artifact in this folder. That being said, if there is a way to avoid deleting files this way and still satisfy the above case, it'd be good to know. 

I guess we can always just delete the "Artifacts" folder, as a potentially less impactful option.

Reply:
We also prompt the user of the folder deletion in the caller of the CreateFolderIfNotExist function here: https://github.com/Azure/azure-powershell/blob/7c8d702511580458b22c28506b4e7f18110155fa/src/Blueprint/Blueprint/Cmdlets/BlueprintDefinition/ExportAzureRMBlueprint.cs#L62